### PR TITLE
Check fee payer before loading all accounts

### PR DESCRIPTION
**Pull request by buffalu**
#### Problem
All accounts are loaded before checking the fee payer's lamport balance. 

#### Summary of Changes
- As soon as the fee payer is loaded, check the balance and bail out

@jstarry @t-nelson

### CODE REVIEW

**Comment by buffalu** [comment]
@jstarry @t-nelson im not the biggest fan of this diff right now bc this method is ~200 lines right now and hard to make sure it was all done correctly.

here's what i propose:
- i separate account loading and fee payer check into two separate functions w/o this fix.
- merge that.
- rebase this fix and move the fee payer check earlier.

lmk what you think

**Comment by Dayday10** [comment]
> @jstarry @t-nelson im not the biggest fan of this diff right now bc this method is ~200 lines right now and hard to make sure it was all done correctly.
> 
> here's what i propose:
> - i separate account loading and fee payer check into two separate functions w/o this fix.
> - merge that.
> - rebase this fix and move the fee payer check earlier.
> 
> lmk what you think

**Comment by Dayday10** [comment]
I'm not sure what to think

**Review by jstarry — DISMISSED** [review]
I think it's fine as is, a bit hard to review but not that bad.

**Comment by jstarry** [comment]
@t-nelson feel free to merge if you get a chance to look. Otherwise I'll merge in my morning assuming CI passes.

**Comment by t-nelson** [comment]
> @jstarry @t-nelson im not the biggest fan of this diff right now bc this method is ~200 lines right now and hard to make sure it was all done correctly.
> 
> here's what i propose:
> 
>     * i separate account loading and fee payer check into two separate functions w/o this fix.
> 
>     * merge that.
> 
>     * rebase this fix and move the fee payer check earlier.
> 
> 
> lmk what you think

agreed. i'd rather see this done incrementally

**Comment by jstarry** [comment]
CI is passing, I already looked over this carefully, refactors can come later IMO

**Comment by jstarry** [comment]
@buffalu can you split this up into 1) refactor and 2) move up the fee payer check as you suggested? Seems like we won't be merge as is

**Comment by buffalu** [comment]
whew, much easier to read this diff

**Review by jstarry — APPROVED** [review]

**Review by t-nelson — DISMISSED** [review]

File: runtime/src/accounts.rs [hunk 1]
```diff
@@ -298,6 +295,23 @@ impl Accounts {
                                 .unwrap_or_default()
                         };
 
+                        if !validated_fee_payer {
```
[the remark this answers is not in the document]
Comment (buffalu) [reply runtime/src/accounts.rs:298]: 👍

**Pull request merged**